tools/type_extract/include/SvGeneric.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "CppEmitter.h"
+
+#include "slang/ast/types/AllTypes.h"
+
+using SvAliases = std::unordered_map<std::string_view, std::string_view>;
+
+class SvGeneric {
+public:
+    enum class Kind { Struct, Enum, LocalParam };
+    explicit SvGeneric(Kind kind) : kind(kind) {}
+
+    virtual void toCpp(HppFile&, std::string_view, const SvAliases&, bool noSystemC) const = 0;
+
+    inline bool isStruct() const { return kind == Kind::Struct; }
+    inline bool isEnum() const { return kind == Kind::Enum; }
+    inline bool isLocalParam() const { return kind == Kind::LocalParam; }
+
+    virtual ~SvGeneric() = default;
+
+protected:
+    Kind kind;
+
+    static std::string_view resolveAlias(std::string_view typeName, const SvAliases& aliases) {
+        if (auto alias = aliases.find(typeName); alias != aliases.end())
+            return alias->second;
+        return typeName;
+    }
+
+    static bool isCppReserved(std::string_view name) {
+        return std::find(cppReservedKeywords.begin(), cppReservedKeywords.end(), name) !=
+               cppReservedKeywords.end();
+    }
+
+    static constexpr std::array<std::string_view, 97> cppReservedKeywords = {"alignas",
```

Comment:
Might be nice to use CTAD here so you don't have to explicitly count the entries: `static constexpr std::array cppReservedKeyword =`